lodash.js

```diff
@@ -2738,6 +2738,29 @@
     }
 
     /**
+     * The base implementation of `_.unset`.
+     *
+     * @private
+     * @param {Object} object The object to query.
+     * @param {Array|string} path The path of the property to unset.
+     * @returns {boolean} Returns delete operator result.
+     */
+    function baseUnset(object, path) {
+      path = isKey(path, object) ? [path + ''] : toPath(path);
+
+      var index = -1,
+          lastIndex = path.length - 1,
+          key;
+
```

Comment:
Since this is unsetting you can use `object = parent(object, path);` once the path is resolved then:
```js
if (object) {
  delete object[last(path)];
}
```